api/utils/keys/policy_test.go

```diff
@@ -14,85 +14,203 @@ See the License for the specific language governing permissions and
 		}, {
-			desc:              string(PrivateKeyPolicyHardwareKeyTouch),
-			errIn:             NewPrivateKeyPolicyError(PrivateKeyPolicyHardwareKeyTouch),
-			expectIsKeyPolicy: true,
-			expectKeyPolicy:   PrivateKeyPolicyHardwareKeyTouch,
+			requiredPolicy:   keys.PrivateKeyPolicyHardwareKeyTouchAndPIN,
+			verifiedPolicies: hardwareKeyTouchAndPINPolicies,
+		},
+	} {
+		t.Run(string(tc.requiredPolicy), func(t *testing.T) {
+			for _, keyPolicy := range privateKeyPolicies {
+				if keys.IsRequiredPolicyMet(tc.requiredPolicy, keyPolicy) {
+					require.Contains(t, tc.verifiedPolicies, keyPolicy, "Policy %q does not meet %q but IsRequirePolicyMet(%v, %v) returned true", keyPolicy, tc.requiredPolicy, tc.requiredPolicy, keyPolicy)
+				} else {
+					require.NotContains(t, tc.verifiedPolicies, keyPolicy, "Policy %q does meet %q but IsRequirePolicyMet(%v, %v) returned false", keyPolicy, tc.requiredPolicy, tc.requiredPolicy, keyPolicy)
+				}
+			}
+		})
+	}
+}
+
+func TestGetPolicyFromSet(t *testing.T) {
+	testCases := []struct {
+		name            string
+		policySet       []keys.PrivateKeyPolicy
+		expectSetPolicy keys.PrivateKeyPolicy
+	}{
+		{
+			name: "none",
+			policySet: []keys.PrivateKeyPolicy{
+				keys.PrivateKeyPolicyNone,
+				keys.PrivateKeyPolicyNone,
+			},
+			expectSetPolicy: keys.PrivateKeyPolicyNone,
+		}, {
+			name: "hardware key policy",
+			policySet: []keys.PrivateKeyPolicy{
+				keys.PrivateKeyPolicyNone,
+				keys.PrivateKeyPolicyHardwareKey,
+			},
+			expectSetPolicy: keys.PrivateKeyPolicyHardwareKey,
+		}, {
+			name: "touch policy",
+			policySet: []keys.PrivateKeyPolicy{
+				keys.PrivateKeyPolicyNone,
+				keys.PrivateKeyPolicyHardwareKey,
+				keys.PrivateKeyPolicyHardwareKeyTouch,
+			},
+			expectSetPolicy: keys.PrivateKeyPolicyHardwareKeyTouch,
+		}, {
+			name: "pin policy",
+			policySet: []keys.PrivateKeyPolicy{
+				keys.PrivateKeyPolicyNone,
+				keys.PrivateKeyPolicyHardwareKey,
+				keys.PrivateKeyPolicyHardwareKeyPIN,
+			},
+			expectSetPolicy: keys.PrivateKeyPolicyHardwareKeyPIN,
+		}, {
+			name: "touch policy and pin policy",
+			policySet: []keys.PrivateKeyPolicy{
+				keys.PrivateKeyPolicyNone,
+				keys.PrivateKeyPolicyHardwareKey,
+				keys.PrivateKeyPolicyHardwareKeyPIN,
+				keys.PrivateKeyPolicyHardwareKeyTouch,
+			},
+			expectSetPolicy: keys.PrivateKeyPolicyHardwareKeyTouchAndPIN,
+		}, {
+			name:            "touch and pin policy",
+			policySet:       privateKeyPolicies,
+			expectSetPolicy: keys.PrivateKeyPolicyHardwareKeyTouchAndPIN,
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			requiredPolicy, err := keys.GetPolicyFromSet(tc.policySet)
+			require.NoError(t, err)
+			require.Equal(t, tc.expectSetPolicy, requiredPolicy)
+
+			// reversing the policy set shouldn't change the output
+			for i, j := 0, len(tc.policySet)-1; i < j; i, j = i+1, j-1 {
+				tc.policySet[i], tc.policySet[j] = tc.policySet[j], tc.policySet[i]
+			}
```

Comment:
```suggestion
			slices.Reverse(tc.policySet)
```

If you want to backport, there's also /x/exp/slices.Reverse.